File: submissions/1019-squares-of-a-sorted-array/solution.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int left = 0;
        int right = nums.size() - 1;
        vector<int> result(nums.size()); 
        int index = nums.size() - 1;    

        while (left <= right) {
            int lel = nums[left] * nums[left];
            int rel = nums[right] * nums[right];

            if (lel > rel) {
                result[index] = lel;
                left++;
            } else {
                result[index] = rel;
                right--;
            }
            
            index--; 
        }

        return result;
    }
};

File: submissions/1019-squares-of-a-sorted-array/solution.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left = 0
        right = len(nums) - 1
        index = right
        
        numsSorted = [0] * len(nums)
        
        while left <= right:
            leftSquare = nums[left] ** 2
            rightSquare = nums[right] ** 2
            
            if leftSquare > rightSquare:
                numsSorted[index] = leftSquare
                left += 1
            else:
                numsSorted[index] = rightSquare
                right -= 1
            
            index -= 1
            
        return numsSorted
